src/components/common/dropdown/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FC } from "react";
+import { UseFormRegister, FieldValues, ValidationRule } from "react-hook-form";
+import { GenderType } from "@/state/features/user";
+import styles from "./Dropdown.module.css";
+
+type DropdownProps = {
+  label: string;
+  options: Array<GenderType>;
+  stylesName: string;
+  register: UseFormRegister<FieldValues>;
+  required: ValidationRule<boolean>;
+  onChange?: () => void;
+};
+
+const Dropdown: FC<DropdownProps> = ({
+  label,
+  options,
+  stylesName = "",
+  register,
+  required,
+  onChange,
+}) => (
+  <div className={`${styles.inputContainer} flex-column`}>
+    <label htmlFor={label}>{label}</label>
+    <select
+      id={label}
+      className={styles[stylesName]}
+      defaultValue=""
+      {...register(label, { required, onChange })}>
+      <option value="" hidden>
+        Select your gender
```

Comment:
Maybe you could let this component more general and use it for different cases and not only to choose gender. It will receive everything from props.

Reply:
I agree, I have added a `defaultOption` as prop